src/upload/src/Upload.tsx

```diff
@@ -312,21 +314,44 @@ export default defineComponent({
     }
     function handleFileAddition (files: FileList | null, e?: Event): void {
       if (!files) return
-      Array.from(files).forEach((file) => {
-        const fileInfo: FileInfo = {
+      const { onBeforeUpload } = props
+      const fileInfoList: FileInfo[] = Array.from(files).map((file) => {
```

Comment:
```suggestion
      const fileInfoList: FileInfo[] = files.map((file) => {
```

不需要 Array.from，map 已经完成了这个功能